homeassistant/components/watchyourlan/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for WatchYourLAN integration."""
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_SSL
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.httpx_client import get_async_client
+
+from .const import DOMAIN
+from .options_flow import WatchYourLANOptionsFlowHandler
+
+# Schema for the user setup form
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_PORT, default=8840): int,
+        vol.Optional(CONF_SSL, default=False): bool,
+        vol.Optional("update_interval", default=5): vol.All(
+            vol.Coerce(int), vol.Range(min=1, max=60)
+        ),
+    }
+)
+
+
+async def validate_input(hass, data):
+    """Validate the user input allows us to connect to the API."""
+    proto = "https" if data[CONF_SSL] else "http"
+    target = f"{proto}://{data[CONF_HOST]}:{data[CONF_PORT]}"
+
+    async with get_async_client(hass) as client:
+        response = await client.get(f"{target}/api/all")
+        if response.status_code != 200:
+            raise CannotConnect
+    return {"title": "WatchYourLAN", "url": target}
```

Comment:
All logic that is required to connect to a device or service should be put in a library hosted on pypi